base/shared/src/main/scala/scalaz/data/IStream.scala

```diff
@@ -0,0 +1,22 @@
+package scalaz
+package data
+
+import scalaz.control.Inf
+
+
+trait IStreamModule {
+  type IStream[A]
+
+  def scons[A](a: =>A)(as: =>Id[IStream[A]]): IStream[A]
+  def uncons[A](as: =>IStream[A]): (A, Inf[IStream[A]])
+}
+
+private[data] object IStreamImpl extends IStreamModule { self =>
+  type IStream[A] = data.Cofree[Id, A]
```

Comment:
This representation of `IStream` will always evaluate at least one element. Was that intended?

Reply:
We could wrap it in `Inf`, that'd be an easy change, and probably more accurate